unittests/basic_tests.py

```diff
@@ -0,0 +1,149 @@
+# * distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+#* ******************************************************************************
+
+import contextlib
+import io
+import os
+from os import path
+import shutil
+import sys
+import unittest
+
+import yaml
+try:
+	from yaml import CSafeLoader as SafeLoader
+except ImportError:
+	from yaml import SafeLoader
+
+MY_PATH = path.abspath(__file__)
+MY_DIR = path.dirname(MY_PATH)
+
+sys.path.append(path.join(MY_DIR,'../src'))
+from yaml_runner import YamlRunner
+
+YAMLRUNNER_PATH = path.abspath(path.join(MY_DIR,'../source/yaml_runner.py'))
+
+class BasicTest(unittest.TestCase):
+    workspace = path.join(MY_DIR,'workspace')
+    test_config_path = path.join(MY_DIR,'../examples/test_config.yml')
+    test_config_file = open(test_config_path, 'r', encoding='utf-8')
+    test_dict = yaml.load(test_config_file, SafeLoader)
+    test_configs = {'path': test_config_path,
+                    'dict' : test_dict,
+                    'file handle' : test_config_file}
+    stderr_log = None
+    stdout_log = None
+    runner = YamlRunner('BasicTest')
+
+    @classmethod
+    def setUpClass(cls):
+        if path.exists(cls.workspace):
+            shutil.rmtree(cls.workspace)
+        os.mkdir(cls.workspace)
+
+    @classmethod
+    def setUp(cls):
+        cls.stderr_log = io.StringIO()
+        cls.stdout_log = io.StringIO()
+
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        shutil.rmtree(cls.workspace)
+        cls.test_config_file.close()
+        return super().tearDownClass()
+
+    @classmethod
+    def tearDown(cls) -> None:
+        cls.stdout_log.close()
+        cls.stderr_log.close()
+
+    def _run_yaml_runner(self,**kwargs):
+        with contextlib.redirect_stdout(self.stdout_log), contextlib.redirect_stderr(self.stderr_log):
+            try:
+                stdout, stderr, exit_code = self.runner.run(**kwargs)
+            except SystemExit as e:
+                exit_code = e.code
+                self.stdout_log.seek(0)
+                self.stderr_log.seek(0)
+                stdout = self.stdout_log.read()
+                stderr = self.stderr_log.read()
+        return stdout, stderr, exit_code
+
+    def test_1_no_config(self):
+        stdout, stderr, exit_code = self._run_yaml_runner(args=['--help'])
+        self.assertEqual(exit_code, 2, 'Test the exit code is zero')
+        self.assertIn('-c', stderr, 'Test the -c option is in the stderr')
+        self.assertIn('--config', stderr, 'Test the --config option is in the stderr')
+
+    def test_2_help_with_config(self):
+        for config_type, config in self.test_configs.items():
+            self._test_help_with_config(config, config_type)
+
+    def _test_help_with_config(self, config, config_type):
```

Comment:
rename the helper methods so it's less confusing
assert_hekp_with_config might work maybe?